src/util/mmap_impl_win32.h

```diff
@@ -101,15 +90,23 @@ inline static DWORD get_system_granuality() {
 }
 
 inline static bool set_file_len(void *handle, size_t size) {
-    LARGE_INTEGER info_size;
-    info_size.QuadPart = static_cast<__int64>(size);
-    FILE_END_OF_FILE_INFO info = {info_size};
-
-    //Makes minimum supported platform Windows Vista
-    return static_cast<bool>(SetFileInformationByHandle(handle,
-                                                        FileEndOfFileInfo,
-                                                        &info,
-                                                        sizeof(info)));
+  BY_HANDLE_FILE_INFORMATION fileInfo = {0};
+  if (GetFileInformationByHandle(handle, &fileInfo) == 0) {
+    return false;
+  }
+
+  LARGE_INTEGER info_size;
+  info_size.LowPart = fileInfo.nFileSizeLow;
+  info_size.HighPart = fileInfo.nFileSizeHigh;
+  if (info_size.QuadPart >= size) {
```

Comment:
Btw there is another check that check out whether we extend above existing file size using `size_` member
Should it be then removed there?

Reply:
In the current impl, it's possible to remove this and still be correct, yep.